test/check.sh

```diff
@@ -518,6 +518,20 @@ it_can_check_with_tag_filter_with_bogus_ref() {
   "
 }
 
+it_can_check_with_tag_filter_with_replaced_tags_from_cache() {
+
+  local repo=$(init_repo)
+  local ref1=$(make_commit_to_branch $repo branch-a)
+  local ref2=$(make_annotated_tag $repo "staging" "tag branch-a")
+  local cache_repo=$(make_cache_repo $repo)
+  local ref3=$(make_commit_to_branch $repo branch-a)
+  local ref4=$(make_annotated_tag $repo "staging" "tag branch-a")
+
+  check_uri_with_tag_filter $cache_repo "staging" | jq -e "
```

Comment:
I wonder if you could avoid the `make_cache_repo` by just running the check twice?

Something like:

```sh
  local repo=$(init_repo)
  local ref1=$(make_commit_to_branch $repo branch-a)
  local ref2=$(make_annotated_tag $repo "staging" "tag branch-a")

  # see that the tag is initially ref2
  check_uri_with_tag_filter $cache_repo "staging" | jq -e "
    . == [{ref: \"staging\", commit: \"$ref2\"}]
  "

  local ref3=$(make_commit_to_branch $repo branch-a)
  local ref4=$(make_annotated_tag $repo "staging" "tag branch-a")

  # see that it is now ref3
  check_uri_with_tag_filter $cache_repo "staging" | jq -e "
    . == [{ref: \"staging\", commit: \"$ref3\"}]
  "
```

IMO this is a bit cleaner and more black-box style. wdyt?

Reply:
I didn't think to run it twice. Agree it should be cleaner. Im going to do it

Reply:
fixed